src/uucore/src/lib/features/parser/num_parser.rs

```diff
@@ -230,14 +230,15 @@ impl ExtendedParser for ExtendedBigDecimal {
     fn extended_parse(
         input: &str,
     ) -> Result<ExtendedBigDecimal, ExtendedParserError<'_, ExtendedBigDecimal>> {
-        parse(input, false)
+        parse(input, ParseTarget::Decimal, &[])
     }
 }
 
-fn parse_special_value(
-    input: &str,
+fn parse_special_value<'a>(
```

Comment:
I'm still a Rust noob, but why do you need those lifetimes? (I also wonder if it should be only on input and not allowed_suffixes)

Reply:
Sounds good thanks.